harvard-lts/etd_alma_drs_holding_service
# Alma DRS Holding - Dropbox

**Prepares the DRS Holding MARCXML for Alma and transfers to the Alma dropbox**
* * *

**JIRA Ticket**: https://at-harvard.atlassian.net/browse/ETD-346

# How should this be tested?

Unit testing:
Follow the instructions in the readme that are in this branch: https://github.com/harvard-lts/etd_alma_drs_holding_service/tree/ETD-346?tab=readme-ov-file#testing

Invoking:
1. Find a Mongo record in the `etds` database that has an `indash` parameter and status of ALMA and copy the PQID
2. Put the PQID into the scripts/invoke-task.py message call
3. Update the .env value CONSUME_QUEUE_NAME to be unique (append initials)
4. Using the local setup from the unit testing above (using the readme in this branch), start up the container
5. Exec into the container `docker exec -it etd-alma-drs-holding-service bash`
6. Run the script from inside the container `python3 scripts/invoke-task.py`
7. Verify that a new XML file exists on hegel@/dropbox/vendors/huletd/incoming called AlmaDRSDarkTest_<timestamp>.xml that appeared when you ran the script.

# Test coverage
Yes/No: Are changes in this pull-request covered by:
- unit tests? yes
- integration tests? yes (manual, not container)
